griptape/structures/workflow.py

```diff
@@ -26,13 +26,21 @@ def input_task(self) -> Optional[BaseTask]:
     def output_task(self) -> Optional[BaseTask]:
         return self.order_tasks()[-1] if self.tasks else None
 
+    @property
+    def input_tasks(self) -> list[BaseTask]:
+        return [task for task in self.tasks if not task.parents]
+
+    @property
+    def output_tasks(self) -> list[BaseTask]:
+        return [task for task in self.tasks if not task.children]
```

Comment:
do we need to call `self.order_tasks()` like above?

Reply:
I don't think so? Because we're not making any assumptions about `self.tasks`'s ordering. We're just grabbing the ones that don't have children.